agent/src/beerocks/monitor/monitor_db.h

```diff
@@ -23,7 +23,10 @@ namespace son {
 ////////////////////////////////////////////
 class monitor_sta_node {
 public:
-    monitor_sta_node(const int8_t vap_id_, const std::string mac_) { vap_id = vap_id_, mac = mac_; }
+    monitor_sta_node(const int8_t vap_id_, const std::string &mac_)
+        : vap_id(vap_id_), mac(mac_), m_sta_stats()
```

Comment:
No need to call the default constructor explicitly. I am referring to the explicit call to `m_sta_stats()`
references -
standard: http://eel.is/c++draft/class.base.init#9
godbolt: https://godbolt.org/z/4YEwWS
Also apply for more default initialized members along the code.

Reply:
cppcheck actually complains about uninitialized members in the constructor:
`'/builds/prpl-foundation/prplMesh/agent/src/beerocks/monitor/monitor_db.h: warning: Member variable 'monitor_sta_node::m_sta_stats' is not initialized in the constructor. [uninitMemberVar]    monitor_sta_node(const int8_t vap_id_, const std::string mac_) { vap_id = vap_id_, mac = mac_; }`

Reply:
Actually, I think cppcheck is right here.
`m_sta_stats` is of type SStaStats, which comes from `common/beerocks/bwl/include/bwl/mon_wlan_hal_types.h` (and there is an other, different definition in `agent/src/beerocks/monitor/monitor_db.h`).

The struct is defined as follow:
```
struct SStaStats {
    float rx_rssi_watt;
    uint8_t rx_rssi_watt_samples_cnt;
    float rx_snr_watt;
    uint8_t rx_snr_watt_samples_cnt;
    // int8_t   rx_rssi_prev=beerocks::RSSI_INVALID;
    // int8_t   rx_rssi_curr=beerocks::RSSI_INVALID;
    uint16_t tx_phy_rate_100kb;
    // uint16_t tx_phy_rate_100kb_avg;
    // uint16_t tx_phy_rate_100kb_min;
    // uint16_t tx_phy_rate_100kb_acc;
    uint16_t rx_phy_rate_100kb;
    // uint16_t rx_phy_rate_100kb_avg;
    // uint16_t rx_phy_rate_100kb_min;
    // uint16_t rx_phy_rate_100kb_acc;
    uint64_t tx_bytes_cnt;
    uint64_t rx_bytes_cnt;
    uint64_t tx_packets_cnt;
    uint64_t rx_packets_cnt;
    uint32_t tx_packets;
    uint32_t tx_bytes;
    uint32_t rx_packets;
    uint32_t rx_bytes;
    uint32_t retrans_count;
    // uint8_t  tx_load_percent_curr=0;
    // uint8_t  tx_load_percent_prev=0;
    // uint8_t  rx_load_percent_curr=0;
    // uint8_t  rx_load_percent_prev=0;
};
```

If you don't initialize m_sta_stats explicitly, all of these will be default-initialized, so they will have a random value.

I adapted your example on godbolt: https://godbolt.org/z/LWmccS

Reply:
yes, I you are correct.
No reason to default initialize this struct unless specifically requested.
My mistake.
standard: http://eel.is/c++draft/dcl.init#7.3